runtime/src/accounts_db.rs

```diff
@@ -5487,14 +5487,14 @@ impl AccountsDb {
         is_startup: bool,
     ) -> Result<(Hash, u64), BankHashVerificationError> {
         if !use_index {
-            let accounts_cache_and_ancestors = if can_cached_slot_be_unflushed {
-                Some((&self.accounts_cache, ancestors))
+            let ancestors = if can_cached_slot_be_unflushed {
+                Some(ancestors)
             } else {
                 None
             };
```

Comment:
Future refactorings can use [`bool::then()`](https://doc.rust-lang.org/std/primitive.bool.html#method.then), which may simplify the code. I.e.

```rust
let ancestors = can_cached_slot_be_unflushed.then(|| ancestors);
```